mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -236,19 +423,32 @@ def __init__(
                 "The custom encode_queries and encode_corpus functions of the model will be used"
             )
             self.retriever = DenseRetrievalExactSearch(retriever, **kwargs)
+        elif is_cross_encoder_compatible(retriever):
+            logger.info(
+                "The custom predict function of the model will be used if not a SentenceTransformer CrossEncoder"
+            )
+            self.retriever = Reranker(retriever, **kwargs)
         else:
             self.retriever = DenseRetrievalExactSearch(DRESModel(retriever), **kwargs)
         self.k_values = k_values
-        self.top_k = max(k_values)
+        self.top_k = (
+            max(k_values) if "top_k" not in kwargs else kwargs["top_k"]
+        )  # can lower it for reranking
+        self.top_k_first_stage = (
+            max(k_values)
+            if "top_k_first_stage" not in kwargs
+            else kwargs["top_k_first_stage"]
+        )
         self.score_function = score_function
 
     def __call__(
-        self, corpus: dict[str, dict[str, str]], queries: dict[str, str], **kwargs
+        self, corpus: dict[str, dict[str, str]], queries: dict[str, str]
     ) -> dict[str, dict[str, float]]:
         if not self.retriever:
             raise ValueError("Model/Technique has not been provided!")
+
         return self.retriever.search(
-            corpus, queries, self.top_k, self.score_function, **kwargs
+            corpus, queries, self.top_k, self.score_function, self.top_k_first_stage
```

Comment:
We now need to differentiate `top_k` in the first stage models and `top_k` for the rerankers